### DISCUSSION TITLE
How to see all gene expressions in all clusters in scRNA-seq?

### DISCUSSION THREAD

**Discussion by CicySue** [satijalab/seurat#7544] [General]
Hey, I'm stuck in this step. After UMAP, I don't know how to see all gene expressions in all clusters. All tutorials from websites are all telling you how to find marker genes in every cluster. But I want to get a file of  the expression of all genes in all clusters. And ideally the row names are genes and the column names are clusters.

**Comment by mbm2224** [comment]
Have you tried aggregating gene expression? You can try using [this function](https://satijalab.org/seurat/reference/aggregateexpression), and then writing out the count matrix as a csv or mtx file. If the idents are clusters, then it will aggregate cells to cluster, so clusters will be column and genes are rows.

Out of curiosity, may I ask why you want this information and what you plan to do with it?

**Comment by CicySue** [comment]
thank u sooooooo much! Because we want to see the difference among
clusters, but the heatmap didn’t do a god job cause it’s too messy

mbm2224 ***@***.***>于2023年7月11日 周二09:48写道：
